drivers/peci/peci_npcx.c

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2022 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_peci
+
+#include <errno.h>
+#include <soc.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/peci.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(peci_npcx, CONFIG_PECI_LOG_LEVEL);
+
+BUILD_ASSERT(IS_ENABLED(CONFIG_PECI_INTERRUPT_DRIVEN),
+	     "CONFIG_PECI_INTERRUPT_DRIVEN must be enabled");
+
+#define PECI_TIMEOUT_MS K_MSEC(300)
+
+struct peci_npcx_config {
+	/* peci controller base address */
+	struct peci_reg *base;
+	struct npcx_clk_cfg clk_cfg;
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct peci_npcx_data {
+	struct k_sem trans_sync_sem;
+	struct k_sem lock;
```

Comment:
Is this lock needed?  Will multiple threads call into the PECI layer?

Reply:
IMHO, we cannot limit  and expect how the application using PECI APIs.
That's why I add the lock to meet possible use cases.

Reply:
The other PECI drivers don't protect against multiple threads, but I won't block this PR.